From 4f4d68a9b772d17a29d1f1efacbf877c3ced857c Mon Sep 17 00:00:00 2001
From: Yehuda Yitschak <yehuday@marvell.com>
Date: Sun, 21 Feb 2016 13:13:47 +0200
Subject: [PATCH 0062/1345] cp110: xhci: add support for usb ports for cp110

commit  23c01e0b71085d28854d64c7dcf0ba166563714e from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

- add basic xhci-plat entries to cp110 dtsi file
- enabled the relevant usb ports in armada-7040 board setups
- enable mvebu xhci plat driver

Change-Id: Ie49ada91c572a4045422b0f9cf83c5a8e7a6df0b
Signed-off-by: Yehuda Yitschak <yehuday@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../boot/dts/marvell/armada-7040-rz-db-nand.dts    |    6 ++++++
 .../boot/dts/marvell/armada-7040-rz-db-router.dts  |    6 ++++++
 arch/arm64/boot/dts/marvell/armada-cp110.dtsi      |   18 ++++++++++++++++++
 3 files changed, 30 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-7040-rz-db-nand.dts b/arch/arm64/boot/dts/marvell/armada-7040-rz-db-nand.dts
index 544b303..29d8222 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-rz-db-nand.dts
+++ b/arch/arm64/boot/dts/marvell/armada-7040-rz-db-nand.dts
@@ -60,6 +60,12 @@
 			sata: sata@540000 {
 				status = "okay";
 			};
+			usb3h0: usb3@500000 {
+				status = "okay";
+			};
+			usb3h1: usb3@510000 {
+				status = "okay";
+			};
 		};
 	};
 };
diff --git a/arch/arm64/boot/dts/marvell/armada-7040-rz-db-router.dts b/arch/arm64/boot/dts/marvell/armada-7040-rz-db-router.dts
index 74e75d0..dab592b 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-rz-db-router.dts
+++ b/arch/arm64/boot/dts/marvell/armada-7040-rz-db-router.dts
@@ -60,6 +60,12 @@
 			sata: sata@540000 {
 				status = "okay";
 			};
+			usb3h0: usb3@500000 {
+				status = "okay";
+			};
+			usb3h1: usb3@510000 {
+				status = "okay";
+			};
 		};
 	};
 };
diff --git a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
index 12acaf7..bbb1fab 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
@@ -39,3 +39,21 @@ sata: sata@540000 {
 	port_base = <0x10000>;
 	port_offset = <0x10000>;
 };
+
+usb3h0: usb3@500000 {
+	compatible = "generic-xhci";
+	reg = <0x500000 0x4000>;
+	dma-coherent;
+	interrupts = <GIC_SPI 62 IRQ_TYPE_LEVEL_HIGH>;
+	clocks = <&gateclk 22>, <&gateclk 16>;
+	status = "disabled";
+};
+
+usb3h1: usb3@510000 {
+	compatible = "generic-xhci";
+	reg = <0x510000 0x4000>;
+	dma-coherent;
+	interrupts = <GIC_SPI 61 IRQ_TYPE_LEVEL_HIGH>;
+	clocks = <&gateclk 23>, <&gateclk 16>;
+	status = "disabled";
+};
-- 
1.7.9.5

